From b1ddfdd85cbbb6f1e7a47754b7b0a87784bc6772 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Wed, 18 Feb 2015 12:40:32 +0100
Subject: [PATCH 12/16] Remove configure directives that do not work and are
 important only for upstream

---
 configure.ac | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index cbf9623..9954471 100644
--- a/configure.ac
+++ b/configure.ac
@@ -159,11 +159,6 @@ LOGDIR=${localstatedir}/log/cluster
 CLUSTERVARRUN=${localstatedir}/run/cluster
 CLUSTERDATA=${datadir}/cluster
 
-AC_PYTHON_MODULE(suds, 1)
-AC_PYTHON_MODULE(pexpect, 1)
-AC_PYTHON_MODULE(pycurl, 1)
-AC_PYTHON_MODULE(requests, 1)
-
 ## path to 3rd-party binaries
 AC_PATH_PROG([IPMITOOL_PATH], [ipmitool], [/usr/bin/ipmitool])
 AC_PATH_PROG([AMTTOOL_PATH], [amttool], [/usr/bin/amttool])
-- 
1.9.3

