diff -urN new/fencemanperl.mk old/fencemanperl.mk
--- new/make/fencemanperl.mk	1970-01-01 01:00:00.000000000 +0100
+++ old/make/fencemanperl.mk	2015-02-18 23:25:09.859528379 +0100
@@ -0,0 +1,9 @@
+%.8: $(TARGET) $(top_srcdir)/fence/agents/lib/fence2man.xsl
+	set -e && \
+		perl $(TARGET) -o metadata > .$@.tmp && \
+	xmllint --noout --relaxng $(top_srcdir)/fence/agents/lib/metadata.rng .$@.tmp && \
+	xsltproc $(top_srcdir)/fence/agents/lib/fence2man.xsl .$@.tmp > $@
+	xsltproc $(top_srcdir)/fence/agents/lib/fence2wiki.xsl .$@.tmp | grep -v '<?xml' > $(@:%.8=%.wiki)
+
+clean-man:
+	rm -f *.8 .*.8.tmp *.wiki
