From 767f6cefdb5f617adb95c31cf7b3e173cb502634 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Thu, 19 Feb 2015 10:08:11 +0100
Subject: [PATCH 1/2] fence_scsi: Fix cleaning of manual pages as is
 distributed

---
 fence/agents/scsi/Makefile.am | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fence/agents/scsi/Makefile.am b/fence/agents/scsi/Makefile.am
index 32f6549..943b174 100644
--- a/fence/agents/scsi/Makefile.am
+++ b/fence/agents/scsi/Makefile.am
@@ -17,6 +17,9 @@ dist_man_MANS		= $(TARGET).8
 
 include $(top_srcdir)/make/fencebuild.mk
 
+clean-man:
+	echo
+
 clean-local:
 	rm -f $(TARGET)
 
-- 
1.9.3

