From 4e791db7f51a32a6a9fa4b48544f43d7d2c4d4ce Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Thu, 26 Feb 2015 10:29:55 +0100
Subject: [PATCH 1/2] fence_kdump: Fix problem in metadata

---
 fence/agents/kdump/fence_kdump.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fence/agents/kdump/fence_kdump.c b/fence/agents/kdump/fence_kdump.c
index 5cea11d..29abbda 100644
--- a/fence/agents/kdump/fence_kdump.c
+++ b/fence/agents/kdump/fence_kdump.c
@@ -233,7 +233,6 @@ do_action_metadata (const char *self)
 
     fprintf (stdout, "\t<parameter name=\"action\" unique=\"0\" required=\"0\">\n");
     fprintf (stdout, "\t\t<getopt mixed=\"-o, --action\" />\n");
-    fprintf (stdout, "\t\t<content type=\"string\" default=\"monitor\" />\n");
     fprintf (stdout, "\t\t<content type=\"string\" default=\"off\" />\n");
     fprintf (stdout, "\t\t<shortdesc lang=\"en\">%s</shortdesc>\n",
              "Fencing action");
-- 
1.9.3

