From cda34baa19dfe8e0fa55ff0eef99614639502cea Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Thu, 26 Mar 2015 13:55:44 +0100
Subject: [PATCH 1/2] [test] Test if XML metadata are valid after fence2rng.xsl

---
 make/agentpycheck.mk | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/make/agentpycheck.mk b/make/agentpycheck.mk
index 291e242..ba573f9 100644
--- a/make/agentpycheck.mk
+++ b/make/agentpycheck.mk
@@ -3,7 +3,7 @@ TEMPFILE:=$(shell mktemp)
 DATADIR:=$(abs_top_srcdir)/tests/data/metadata
 AWK_VAL='BEGIN {store=-1} /name=\"store_path\"/ {store=2} {if (store!=0) {print}; store--}'
 
-check: $(TARGET:%=xml-check.%) $(SYMTARGET:%=xml-check.%) $(TARGET:%=delay-check.%)
+check: $(TARGET:%=xml-check.%) $(SYMTARGET:%=xml-check.%) $(TARGET:%=delay-check.%) $(TARGET:%=rng-check.%)
 
 xml-check.%: %
 	$(eval INPUT=$(subst xml-check.,,$@))
@@ -23,3 +23,9 @@ delay-check.%: %
 	sed 's/\.//' | tail -n 1` -ge 1000 || \
 	PYTHONPATH=$(abs_srcdir)/../lib:$(abs_builddir)/../lib /usr/bin/time -f "%e" \
 	python ./$(INPUT) --delay 0 $(FENCE_TEST_ARGS) --
+
+rng-check.%: %
+	PYTHONPATH=$(abs_srcdir)/../lib:$(abs_builddir)/../lib python ./$(INPUT) -o metadata | \
+	/usr/bin/xsltproc ${abs_top_srcdir}/fence/agents/lib/fence2rng.xsl - | \
+	sed -e 's/ rha:description=/ description=/g' -e 's/ rha:name=/ name=/g' | \
+	xmllint --nsclean --noout -
-- 
1.9.3

